addons/hr_expense/models/hr_expense.py

```diff
@@ -504,6 +506,11 @@ def _compute_is_multiple_currency(self):
         for sheet in self:
             sheet.is_multiple_currency = len(sheet.expense_line_ids.mapped('currency_id')) > 1
 
+    @api.multi
+    def _compute_can_reset(self):
```

Comment:
Just because I don't like stacked 'and' and 'or', could be try something like
```
is_expense_user = self.user_has_groups('hr_expense.group_hr_expense_user')
can_reset = is_expense_user if is_expense_user else sheet.employee_id.user_id == self.env.user
```